### ISSUE TITLE
Nuxt is rendering the previous page 

### ISSUE THREAD

**Issue by just-tom** [nuxt/nuxt.js#6553]
### Version

[v2.10.0](https://github.com/nuxt.js/releases/tag/v2.10.0)

### Reproduction link

[https://github.com/just-tom/more-pwa-test](https://github.com/just-tom/more-pwa-test)

### Steps to reproduce

See the issue here: https://master-7rqtwti-3wuo2xdkxutcm.eu-4.platformsh.site or clone the repo.

1) Go onto the reproduction website
2) Scroll down on homepage
3) Click Catalog on App Bar
4) Click Industrial
5) This Loads the Category for Industrial products & sets currentCategory in the vuex store.
6)Now click the logo to go back to the home page - there is a middleware that set currentCategory to null if we are not heading to a category page.
7) On the homepage inspect the console. See the errors

### What is expected ?

No Errors should be in the console. 
currentCategory obj should only be logged to console when rendering the category page.

### What is actually happening?

When going from the Industrial category to the homepage an error is shown about .name. This is trying to output currentCategory.name as the title on the category page......but we arent on or going to a category page, we are going to home page.

To further provie the point the console.log of current category returns a null value - which is correct. but that console.log is in the category page vue file (ranges/_slug.vue) not in the index, so why is it being called.

### Additional comments?

Basically when going to the homepage its like the previous page is being rendered as well, but the data is not there for it as I remove the currentCategory from VueX in the middleware, so it errors, but the page shouldn't be being re-rendered the previous page.

If you reload the pages manually with the browser there is no error. This only happens when using a nuxt-link on the client side.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9893">#c9893</a>)</em></sub></div>

**Comment by pimlie** [comment]
Hmm, I dont see an app bar in the reproduction?

**Comment by just-tom** [comment]
@pimlie You have to scroll down slightly on the home page. Thanks

**Comment by just-tom** [comment]
At the moment I have fixed this by implementing a "previous category" type element, but its far from ideal
